### PR TITLE
[WIP]: Added more cases for the tests. Fixes #115. Added Image Info to the source image …

### LINKED ISSUE #115: Move "thread" param out of URL params and into app config
It was left from the beginning , it's already private but it's still on the list of URL params when it should be a config level thing.

### PR DESCRIPTION
**Pull request by baamenabar**
…(needed for calculations).

There are double the tests, but the later half is deactivater

### CODE REVIEW

File: src/Core/Entity/Image/InputImage.php [hunk 1]
```diff
@@ -19,6 +20,9 @@ class InputImage
     /** @var string */
     protected $sourceImageMimeType;
 
+    /** @var array Associative array that holds basic image info. Consider in the future create an imageInfo class. */
+    protected $imageInfo;
```
Comment (sadok-f) [inline src/Core/Entity/Image/InputImage.php]: Yes definitely, we need to move this property into a sperate class, call it Core\Entity\Image\ImageMetaInfo.php 
and move the method related to this new class.

File: src/Core/Processor/ImageProcessor.php [hunk 2]
```diff
@@ -156,11 +171,33 @@ protected function generateSize(OutputImage $outputImage): array
...
-            $extent = '';
-        }
 
         return [$size, $extent, $gravity];
     }
+
+    /**
+     * @return array
+     */
+    protected function getSourceImageDimensions()
```
Comment (sadok-f) [inline src/Core/Processor/ImageProcessor.php]: where those functions `getSourceImageDimensions ` and `getSourceImageInfo ` called ?

Comment (baamenabar) [reply src/Core/Processor/ImageProcessor.php]: These are called in ImageProcessor

File: src/Core/Entity/Image/InputImage.php (hunk 1 above)
Comment (baamenabar) [reply src/Core/Entity/Image/InputImage.php]: Awesome, will do

File: src/Core/Entity/Image/InputImage.php [hunk 3]
```diff
@@ -132,4 +136,95 @@ public function getSourceImageMimeType(): string
...
+        return $imageDetails;
+    }
+
+    /**
+     * @param string $commandStr
+     *
+     * @return array
+     * @throws \Exception
+     */
+    public static function execute(string $commandStr): array
```
[the remark this answers is not in the document]
Comment (baamenabar) [reply src/Core/Entity/Image/InputImage.php]: do I add an instance of processor in here?  
Or could I add the executeInfoComand method inside the ImageMetaInfo class and just intanciate the class here?.... hmmm actually I only need it in the ImageProcessor class.

I know what I'll do. Push later

**Comment by baamenabar** [comment]
Alright! I'm done with this PR, I have more tests, but they are failing and I need to fix the functionallity. I'll do that in another PR. I've done enough here. And all works same as before.

We do need to fix all cases for the image identification (minor bugs there).

Also need to create an abstract Executor class for the command line executions and binary paths.

**Pull request merged**